### PR TITLE
Make installable via cocoapods on iOS and added support for promises, updating to use non deprecated methods.

### PR DESCRIPTION
**Pull request by robertbarclay**

### CODE REVIEW

File: android/src/main/java/com/github/wumke/RNImmediatePhoneCall/RNImmediatePhoneCallModule.java [hunk 1]
```diff
@@ -34,35 +35,59 @@ public String getName() {
...
-            call();
-        } else {
-            ActivityCompat.requestPermissions(getCurrentActivity(),
+        if (number == null || number.isEmpty()) {
+            promise.reject(new JSApplicationIllegalArgumentException("Invalid number: " + url));
+            return;
+        }
+
+        try {
+            Bool hasPermission = ContextCompat.checkSelfPermission(reactContext.getApplicationContext(),
```
Comment (TylerWhitman) [inline android/src/main/java/com/github/wumke/RNImmediatePhoneCall/RNImmediatePhoneCallModule.java]: There isnt Bool in java, should be Boolean. https://docs.oracle.com/javase/8/docs/api/index.html?java/lang/Boolean.html

Comment (TylerWhitman) [reply android/src/main/java/com/github/wumke/RNImmediatePhoneCall/RNImmediatePhoneCallModule.java]: Unless this is some custom object?

File: android/src/main/java/com/github/wumke/RNImmediatePhoneCall/RNImmediatePhoneCallModule.java [hunk 2]
```diff
@@ -34,35 +35,59 @@ public String getName() {
...
         }
     }
 	
 	@SuppressLint("MissingPermission")
-    private static void call() {
-        String url = "tel:" + RNImmediatePhoneCallModule.number;
-        Intent intent = new Intent(Intent.ACTION_CALL, Uri.parse(url));
-        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-        rnImmediatePhoneCallModule.reactContext.startActivity(intent);
+    private static void call(String number, Promise promise, Activity activity) {
```
Comment (TylerWhitman) [inline android/src/main/java/com/github/wumke/RNImmediatePhoneCall/RNImmediatePhoneCallModule.java]: Do you need to resolve the promise?

File: index.js [hunk 3]
```diff
@@ -1,9 +1,49 @@
...
-  immediatePhoneCall: function(number) {
-        NativeModules.RNImmediatePhoneCall.immediatePhoneCall(number);
+  immediatePhoneCall: async function(number, title = defaultTitle, message = defaultMessage, buttonNegative = 'Cancel', buttonPositive = 'OK') {
+    if (isAndroid) {
+      try {
+        let permission = await PermissionsAndroid.check(PermissionsAndroid.PERMISSIONS.CALL_PHONE);
+        if(!permission) {
+          let granted = await requestCallPermission(title, message, buttonNegative, buttonPositive);
+          if (granted) {
+            return NativeModules.RNImmediatePhoneCall.immediatePhoneCall("411");
```
Comment (jimmyfortinx) [inline index.js]: Hardcoded 411 here

File: index.js [hunk 4]
```diff
@@ -1,9 +1,49 @@
...
+          }
+          return Promise.reject("Permission not granted");
+        }
+      } catch (e) {
+        return Promise.reject(e);
+      }
+    }
+    return NativeModules.RNImmediatePhoneCall.immediatePhoneCall("411");
```
Comment (jimmyfortinx) [inline index.js]: Hardcoded 411 here too

File: index.js [hunk 5]
```diff
@@ -1,9 +1,49 @@
...
+        message,
+        buttonNegative,
+        buttonPositive
+      },
+    );
+    if (granted === PermissionsAndroid.RESULTS.GRANTED) {
+      return true;
+    } else {
+      return false;
+    }
```
Comment (jimmyfortinx) [inline index.js:43]:
```suggestion
    return granted === PermissionsAndroid.RESULTS.GRANTED;
```

File: index.js [hunk 6]
```diff
@@ -1,9 +1,49 @@
...
+  } catch (err) {
+    return false;
```
Comment (jimmyfortinx) [inline index.js:45]: I'm not sure about removing the error... This could hide something important to the user that will use this lib.

**Review by jimmyfortinx — CHANGES_REQUESTED** [review]

**Pull request closed without merging**